.github/autobuild/android.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+set -eu
+
+COMMANDLINETOOLS_VERSION=6858069
+ANDROID_NDK_VERSION=r21d
+ANDROID_PLATFORM=android-30
+ANDROID_BUILD_TOOLS=30.0.2
+AQTINSTALL_VERSION=2.0.6
+QT_VERSION=5.15.2
+
+# Only variables which are really needed by sub-commands are exported.
+# Definitions have to stay in a specific order due to dependencies.
+QT_BASEDIR="/opt/Qt"
+ANDROID_BASEDIR="/opt/android"
+export ANDROID_SDK_ROOT="${ANDROID_BASEDIR}/android-sdk"
+COMMANDLINETOOLS_DIR="${ANDROID_SDK_ROOT}"/cmdline-tools/latest/
+ANDROID_NDK_ROOT="${ANDROID_BASEDIR}/android-ndk"
+ANDROID_NDK_HOST="linux-x86_64"
+ANDROID_SDKMANAGER="${COMMANDLINETOOLS_DIR}/bin/sdkmanager"
+export JAVA_HOME="/usr/lib/jvm/java-8-openjdk-amd64/"
+export PATH="${PATH}:${ANDROID_SDK_ROOT}/tools"
+export PATH="${PATH}:${ANDROID_SDK_ROOT}/platform-tools"
+
+if [[ ! ${jamulus_buildversionstring:-} =~ [0-9]+\.[0-9]+\.[0-9]+ ]]; then
+    echo "Environment variable jamulus_buildversionstring has to be set to a valid version string"
+    exit 1
+fi
+
+setup_ubuntu_dependencies() {
+    export DEBIAN_FRONTEND="noninteractive"
+
+    sudo apt-get -qq update
+    sudo apt-get -qq --no-install-recommends -y install build-essential zip unzip bzip2 p7zip-full curl chrpath \
+        libxkbcommon-x11-0 openjdk-8-jre-headless openjdk-8-jdk-headless
+}
+
+setup_android_sdk() {
+    mkdir -p "${ANDROID_BASEDIR}"
+
+    if [[ -d "${COMMANDLINETOOLS_DIR}" ]]; then
+        echo "Using commandlinetools installation from previous run (actions/cache)"
+    else
+        mkdir -p "${COMMANDLINETOOLS_DIR}"
+        curl -s -o downloadfile "https://dl.google.com/android/repository/commandlinetools-linux-${COMMANDLINETOOLS_VERSION}_latest.zip"
```

Comment:
We should check the md5 hash on downloads in my opinion (but that's also not related).

Reply:
Hmm...
- Regarding download errors (corruptions), we should be safe due to TCP (lost or corrupted packets) and TLS (mangled packets). Both would fail/retry if transmission errors occur.
- Regarding security, I'm not sure what the benefit would be. Where would we get the hash from? From the very Google page as well?

Reply:
Do it once locally and then put it up into the script?

Reply:
Well, but where do I get the hash from in the first place? I would have to download from the very same source, so the only protection benefit would be a benefit over time, i.e. if Google became malicious or a malicious actor could replace the file there.
We aren't handling that attack vector anywhere else either (Qt, choco, pip, Android SDK contents, ...), do we?

I'm not opposed to doing that per-se, but I'd rather have a discussion and do it in all places if we decide to do it. Remember that it will carry a rather big maintenance burden though.

I think the current consensus is:
- Ensure authenticity/integrity of downlods by using HTTPS
- Use trusted download URLs
- Pin versions, especially for user-generated content (e.g. pip, choco) [and elsewhere for reproducibility reasons)

Reply:
Probably this is #1737 then.